### PR TITLE
Disable hoisting for installs in the rushstack repository

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary
Ensures that packages and external dependencies do not have phantom dependencies at build time. Any dependency not expressly declared in package.json (or corrected via pnpmfile) will not be available to projects.

## Details
Updates dependency footprints to ensure that all projects successfully build and execute in the presence of maximum installation strictness.

## How it was tested
rush update --full --purge
rush retest --verbose

### CODE REVIEW

**Review by TheLarkInn — APPROVED** [review]
LGTM

File: build-tests-samples/heft-serverless-stack-tutorial/tsconfig.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
     "module": "commonjs",
     "target": "es2017",
-    "lib": ["es2017"]
+    "lib": ["es2017", "DOM"]
```
Comment (iclanton) [inline build-tests-samples/heft-serverless-stack-tutorial/tsconfig.json:25]: Related?

File: rush.json [hunk 2]
```diff
@@ -1060,13 +1060,13 @@
       "packageName": "@rushstack/module-minifier-plugin",
       "projectFolder": "webpack/module-minifier-plugin",
       "reviewCategory": "libraries",
-      "shouldPublish": false
+      "shouldPublish": true
```
Comment (iclanton) [inline rush.json:1063]: Is the publishing issue fixed?

Comment (iclanton) [reply rush.json:1063]: If we're going to go with `@rushstack/webpack5-module-minifier-plugin`, we should deprecate+tombstone `@rushstack/module-minifier-plugin` and replace it with `@rushstack/webpack4-module-minifier-plugin`

File: build-tests-samples/heft-serverless-stack-tutorial/tsconfig.json (hunk 1 above)
Comment (dmichon-msft) [reply build-tests-samples/heft-serverless-stack-tutorial/tsconfig.json:25]: Yes, `esbuild` needs the typings for the `WebAssembly` namespace, which is provided by the `DOM` lib in TypeScript.

File: rush.json (hunk 2 above)
Comment (dmichon-msft) [reply rush.json]: Moving re-enabling this to a separate PR

**Review by D4N14L — APPROVED** [review]

**Review by iclanton — APPROVED** [review]

**Pull request merged**